packages/react/src/components/Tab/Tab.js

```diff
@@ -105,9 +106,11 @@ export default class Tab extends React.Component {
     onKeyDown: () => {},
   };
 
+  tabId = this.props.id || `tab-${getInstanceId()}`;
```

Comment:
One quick question, we've gotten some feedback around DOM mismatches when using auto-ids and server-side rendering. Curious if there is something we could do here, or if we're just saying if you want to SSR then you should pass in an `id` 👀 

Reference for the `useId` hook for background (totally get we can't use the hook in a class component): https://github.com/carbon-design-system/carbon/blob/master/packages/react/src/internal/useId.js#L8

Reply:
> we're just saying if you want to SSR then you should pass in an `id`
yeah this sounds logical to me, but how have we handled it previously? the auto-id generators can be found throughout the library currently so I thought it was a viable solution here

Reply:
@emyarod sounds great to me, especially for the class components 👍